js/raster_layer_editor.ts

```diff
@@ -0,0 +1,345 @@
+    }
+
+    static styles = [
+        flexStyles,
+        legacyStyles,
+        materialStyles,
+        css`
+            .horizontal-flex > button {
+                height: 28px;
+                width: 28px;
+            }
+        `,
+    ];
+
+    @property({ type: Array }) stretchOptions: Array<SelectOption> = [
+        { label: "Custom", value: "custom" },
+        { label: "1σ", value: "sigma-1" },
+        { label: "2σ", value: "sigma-2" },
+        { label: "3σ", value: "sigma-3" },
+        { label: "90%", value: "percent-90" },
+        { label: "98%", value: "percent-98" },
+        { label: "100%", value: "percent-100" },
+    ];
+    @property({ type: Array }) colorModels: Array<SelectOption> = [
+        { label: "1 band (Grayscale)", value: ColorModel.Gray },
+        { label: "3 bands (RGB)", value: ColorModel.RGB },
+    ];
+    @property({ type: Array }) colorRamps: Array<SelectOption> = [
+        { label: "Gamma", value: ColorRamp.Gamma },
+        { label: "Palette", value: ColorRamp.Palette },
+    ];
+
+    @property({ type: String }) colorModel: string = "";
+    @property({ type: Array }) bandNames: Array<string> = [];
+    @property({ type: Array }) selectedBands: Array<string> = [];
+    @property({ type: String }) stretch: string = this.stretchOptions[0].value;
+    @property({ type: Number }) minValue: number | undefined = 0.0;  // undefined is loading state.
+    @property({ type: Number }) maxValue: number | undefined = 1.0;  // undefined is loading state.
+    @property({ type: Boolean }) minAndMaxValuesLocked: boolean = false;
+    @property({ type: Number }) opacity: number = 1.0;
+    @property({ type: String }) colorRamp: string = ColorRamp.Gamma;
+    @property({ type: Number }) gamma: number = 1.0;
+    @property({ type: Array }) colormaps: Array<string> = [];
+
+    @query('palette-editor') paletteEditor?: PaletteEditor;
+    @query('legend-customization') legendCustomization?: LegendCustomization;
+
+    connectedCallback() {
+        super.connectedCallback()
+        this.colorModel = this.bandNames.length > 1 ? ColorModel.RGB : ColorModel.Gray;
+    }
+
+    getVisualizationOptions(): any {
+        const visOptions = {
+            bands: this.selectedBands,
+            min: this.minValue,
+            max: this.maxValue,
+            opacity: this.opacity,
+        } as any;
+        if (this.colorModel === ColorModel.Gray) {
+            if (this.colorRamp === ColorRamp.Palette) {
+                visOptions.palette = this.paletteEditor?.paletteTokens() ?? [];
+            } else if (this.colorRamp === ColorRamp.Gamma) {
+                visOptions.gamma = this.gamma;
+            }
+        } else if (this.colorModel === ColorModel.RGB) {
+            visOptions.gamma = this.gamma;
+        }
+        return visOptions;
+    }
+
+    render(): TemplateResult {
+        return html`
+            <div class="vertical-flex">
+                <div class="horizontal-flex">
```

Comment:
Wrap the color model radio and band selection elements in a `<fieldset>` for better visual separation:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset